jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -1180,7 +1266,14 @@ ecma_string_get_char_at_pos (const ecma_string_t *string_p, /**< ecma-string */
   lit_utf8_size_t sz = ecma_string_to_utf8_string (string_p, utf8_str_p, buffer_size);
   JERRY_ASSERT (sz == buffer_size);
 
-  ch = lit_utf8_string_code_unit_at (utf8_str_p, buffer_size, index);
+  if (is_ascii)
```

Comment:
If I see this correctly then we can only get here, when chars_p is NULL and that means we have to deal with ECMA_STRING_CONTAINER_HEAP_NUMBER and ECMA_STRING_CONTAINER_UINT32_IN_DESC. This means that this will be always true. I'd change this to a JERRY_ASSERT